packages/esm-patient-common-lib/src/workspaces/workspaces.ts

```diff
@@ -243,11 +244,22 @@ export function changeWorkspaceContext(patientUuid) {
   }
 }
 
-const initialState = { patientUuid: null, openWorkspaces: [], prompt: null };
+const initialState: WorkspaceStoreState = {
+  patientUuid: null,
+  openWorkspaces: [],
+  prompt: null,
+  workspaceWindowState: 'hidden',
+};
 export function getWorkspaceStore() {
   return getGlobalStore<WorkspaceStoreState>('workspace', initialState);
 }
 
+export function updateWorkspaceSizeState(value: WorkspaceWindowState) {
+  const store = getWorkspaceStore();
+  const state = store.getState();
+  store.setState({ ...state, workspaceWindowState: value });
+}
```

Comment:
```suggestion
export function updateWorkspaceSizeState(value: WorkspaceWindowState) {
  getWorkspaceStore().setState((state) => { ...state, workspaceWindowState: value });
}
```

This might be a unistore vs zustand API difference.

Reply:
However, instead of having a separate exported function for this, it might make sense in `useWorkspaces()` to something like:

```ts
const { prompt, openWorkspaces: workspaces, workspaceWindowSize, updateWorkspaceSize  } = useStoreWithActions(getWorkspaceStore(), { updateWorkspaceSize: (state, value} => ({ ...state, workspaceWindowState: value) })})
```

Reply:
The update function doesn't use anything from the hook, so I think it's fine to leave it as a function globally available. Contrast for example with `useOrderBasket`, where the update function actually uses the `useOrderBasket` arguments.

Reply:
This is also consistent with the rest of the workspace API, which follows this pattern, rather than using actions. I have no strong feelings between the two paradigms (actions vs global functions that act on the store), but I think consistency is good.

Reply:
I was thinking of returning the `updateWorkspaceWindowState` from `workspaces.ts` via `useWorkspaces`, which didn't quite fit right.

I implemented it with `useStoreWithActions` for now, so your comment late. Should I revert the changes, or we can go ahead with the same?
Thanks!

Reply:
Yeah now we have this fragmented API where most workspace functions are imported from the library, except updateWorkspaceWindowSize which is returned from the hook for no good reason. Please change it back and remove it from the hook.

Reply:
I understand, pushing the changes

Reply:
Done!